crates/fuel-core/src/service/adapters.rs

```diff
@@ -85,6 +86,155 @@ impl StaticGasPrice {
     }
 }
 
+#[cfg(test)]
+mod arc_gas_price_estimate_tests {
+    #![allow(non_snake_case)]
+
+    use super::*;
+    use proptest::proptest;
+
+    async fn _worst_case__correctly_calculates_value(
```

Comment:
nit: this function is called later on
```suggestion
    async fn worst_case__correctly_calculates_value(
```

Reply:
Yes. It's the actual test code, but I don't like writing it inside the `prop_test` macro because the IDE has a hard time parsing it:
```
proptest! {
    #[test]
    fn worst_case__correctly_calculates_value(
        price: u64,
        starting_height: u32,
        block_horizon in 0..10_000u32,
        percentage: u64
    ) {
        _worst_case__correctly_calculates_value(price, starting_height, block_horizon, percentage);
    }
}
```
So this is my naming convention for that case.